nixos/modules/programs/gnupg.nix

```diff
@@ -123,7 +123,7 @@ in
 
     services.dbus.packages = mkIf (cfg.agent.pinentryFlavor == "gnome3") [ pkgs.gcr ];
 
-    environment.systemPackages = with pkgs; [ cfg.package ];
+    environment.systemPackages = with pkgs; [ cfg.package pinentry ];
```

Comment:
Shouldn't we install the proper pinentry package (i.e. qt, gtk or curses flavour)?

Reply:
Not really sure how to implement that logic

Reply:
How about solving it just like it's done here: https://github.com/NixOS/nixpkgs/blob/7e9b1564e661026f42c09fb6cfa521dd4294c445/nixos/modules/programs/gnupg.nix#L97-L102

Reply:
that's constructing a path, I don't think I can put just a path as an item to: `environment.systemPackages`

Reply:
I'm talking about adding `pinentry.${cfg.agent.pinentryFlavor}` to `systemPackages`. The `pinentry`-package as an output for each flavor.

Reply:
ah, interesting. I misread that line, though it was "${pkgs.pinentry}.${cfg.agent.pinentryFlavor}", as that's more common